### ISSUE TITLE
Problem parsing generic function parameters when function split over multiple lines

### ISSUE THREAD

**Issue by mwoollard** [realm/jazzy#896]
When documenting the following definition the following is parsed fine:
```
public protocol VehiclePositionManager {
    func positions(for region: Observable<LocationRegion?>, stopId: String?, routeIds: [String]?) -> Observable<[VehiclePositionType]>
    
}
```
![jazzy1](https://user-images.githubusercontent.com/150681/31814511-c9b609ce-b581-11e7-937c-fca3bcdde8df.jpeg)

But if the function is broken over several lines it gets truncated in the docs:
```
public protocol VehiclePositionManager {
    func positions(for region: Observable<LocationRegion?>,
                   stopId: String?, 
                   routeIds: [String]?) -> Observable<[VehiclePositionType]>
}
```
![jazzy2](https://user-images.githubusercontent.com/150681/31814535-e0ba196c-b581-11e7-97e7-b87c41cac25a.jpeg)

This problem doesn't appear to happen if the function parameters don't include any generic definitions.

Thanks
Mark

**Comment by johnfairh** [comment]
I think this happens for all multi-line function declarations inside protocols, the generic types look like  a coincidence.

**Comment by johnfairh** [comment]
This turns out to be tough to completely fix.  SourceKitten normally uses the 'body offset' key to know when the declaration ends, which is absent for protocol methods.  A rule like _for functions without 'body offset', use 'length'_ works correctly in the example here, but it turns out 'length' does not include any `where` clause meaning that always gets dropped, even if all on one line.  Not sure if that is an improvement.

With PR #902 as-is, these function definitions show up entirely but all on one long line, which is correct but ugly.  Could add a ghastly rule there, assuming the sourcekitten change [ie. respect multi-line but always drop `where` clause], to say "if protocol func has multiple lines then use the multi-line version, unless it also has generic parameter constraints in which case use the all-on-one-line version".

**Comment by johnfairh** [comment]
I fixed the underlying Swift problem with `where` so this issue should be fixable for real in 4.2/5.0 with a SourceKitten tweak.

**Comment by johnfairh** [comment]
Jazzy has a workaround for this now, prints the function on a single line.  Trying to do the 'real' fix found another Swift bug causing a separate regression: will have to wait another cycle.

**Comment by mwoollard** [comment]
OK thanks for the update

**Comment by johnfairh** [comment]
Fixed in master (with a Swift 5.1+ project)
